sql/codegen_xgboost.go

```diff
@@ -0,0 +1,606 @@
+		return nil
+	}
+}
+
+func fp32Partial(key string, ptrFn func(*xgboostFiller) *float32) func(*map[string][]string, *xgboostFiller) error {
+	return func(a *map[string][]string, r *xgboostFiller) error {
+		// xgParseAttr will ensure the key is existing in map
+		val, _ := (*a)[key]
+		if len(val) != 1 {
+			return xgInvalidAttrValError(key, val)
+		}
+		if fpVal, err := strconv.ParseFloat(val[0], 32); err != nil {
+			return xgInvalidAttrValError(key, val)
+		} else if fpPtr := ptrFn(r); *fpPtr != 0 {
+			return xgDupAttrSettingError(key)
+		} else {
+			*fpPtr = float32(fpVal)
+			delete(*a, key)
+		}
+		return nil
+	}
+}
+
+func boolPartial(key string, ptrFn func(*xgboostFiller) *bool) func(*map[string][]string, *xgboostFiller) error {
+	return func(a *map[string][]string, r *xgboostFiller) error {
+		// xgParseAttr will ensure the key is existing in map
+		val, _ := (*a)[key]
+		if len(val) != 1 {
+			return xgInvalidAttrValError(key, val)
+		}
+		bVal, err := strconv.ParseBool(val[0])
+		if err != nil {
+			return xgInvalidAttrValError(key, val)
+		}
+		bPtr := ptrFn(r)
+		*bPtr = bVal
+		delete(*a, key)
+		return nil
+	}
+}
+
+func strPartial(key string, ptrFn func(*xgboostFiller) *string) func(*map[string][]string, *xgboostFiller) error {
+	return func(a *map[string][]string, r *xgboostFiller) error {
+		// xgParseAttr will ensure the key is existing in map
+		val, _ := (*a)[key]
+		if len(val) != 1 {
+			return xgInvalidAttrValError(key, val)
+		}
+		stringPtr := ptrFn(r)
+		if len(*stringPtr) != 0 {
+			return xgDupAttrSettingError(key)
+		}
+		*stringPtr = val[0]
+		delete(*a, key)
+		return nil
+	}
+}
+
+func sListPartial(key string, ptrFn func(*xgboostFiller) *[]string) func(*map[string][]string, *xgboostFiller) error {
+	return func(a *map[string][]string, r *xgboostFiller) error {
+		// xgParseAttr will ensure the key is existing in map
+		val, _ := (*a)[key]
+		strListPtr := ptrFn(r)
+		if len(*strListPtr) != 0 {
+			return xgDupAttrSettingError(key)
+		}
+		*strListPtr = val
+		delete(*a, key)
+		return nil
+	}
+}
+
+var xgbAttrSetterMap = map[string]func(*map[string][]string, *xgboostFiller) error{
+	// runtime params
+	"run_local": boolPartial("run_local", func(r *xgboostFiller) *bool { return &(r.runLocal) }),
+	"workers":   uIntPartial("workers", func(r *xgboostFiller) *uint { return &(r.WorkerNum) }),
+	"memory":    uIntPartial("memory", func(r *xgboostFiller) *uint { return &(r.MemorySize) }),
+	"cpu":       uIntPartial("cpu", func(r *xgboostFiller) *uint { return &(r.CPUSize) }),
+	// booster params
+	"objective":            strPartial("objective", func(r *xgboostFiller) *string { return &(r.Objective) }),
+	"booster":              strPartial("booster", func(r *xgboostFiller) *string { return &(r.Booster) }),
+	"max_depth":            uIntPartial("max_depth", func(r *xgboostFiller) *uint { return &(r.MaxDepth) }),
+	"num_class":            uIntPartial("num_class", func(r *xgboostFiller) *uint { return &(r.NumClass) }),
```

Comment:
I saw there are too many attr keyword here, are these all necessary? I'm not good at XGBoost, for the experience 
 of TF codegen, the Go code only need to parse the common keyword such as `batch_size`, `verbose` and etc.. into Resolver:
https://github.com/sql-machine-learning/sqlflow/blob/a290673e54c3a128ad6a8dc4d551e632abe0b841/sql/expression_resolver.go#L50-L54

the others can fill into Python code as` train_args`:
https://github.com/sql-machine-learning/sqlflow/blob/a290673e54c3a128ad6a8dc4d551e632abe0b841/sql/codegen.go#L297

Reply:
Because `xgblauncher` accept nested python structure as configure,  I want to do more  work with golang, less work with template. Since long templates is harder to read and maintain than golang.